src/App/Validator/HXL/Metadata/Create.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/**
+ * Ushahidi Set Validator
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2014 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+namespace Ushahidi\App\Validator\HXL\Metadata;
+
+use Ushahidi\Core\Entity\ExportJobRepository;
+use Ushahidi\Core\Entity\HXL\HXLLicenseRepository;
+use Ushahidi\Core\Entity\HXL\HXLMetadataRepository;
+use Ushahidi\Core\Entity\UserRepository;
+use Ushahidi\Core\Tool\Validator;
+
+class Create extends Validator
+{
+    protected $repo;
+    protected $user_repo;
+    protected $license_repo;
+    protected $export_job_repo;
+
+    public function __construct(
+        HXLMetadataRepository $repo,
+        UserRepository $user_repo,
+        HXLLicenseRepository $license_repo,
+        ExportJobRepository $export_job_repo
+    ) {
+        $this->repo = $repo;
+        $this->user_repo = $user_repo;
+        $this->license_repo = $license_repo;
+        $this->export_job_repo = $export_job_repo;
+    }
+    /**
+     * @return array|\Ushahidi\Core\Tool\ArrayValidation
+     */
+    protected function getRules()
+    {
+        return array_merge([
+            'private' => [
+                ['not_empty']
+            ],
+            'dataset_title' => [
+                ['not_empty'],
+                ['min_length', [':value', 2]],
+                ['max_length', [':value', 255]],
+                ['regex', [':value', Validator::REGEX_STANDARD_TEXT]], // alpha, number, punctuation, space
+            ],
+            'organisation' => [
+                ['not_empty'],
+                ['min_length', [':value', 1]],
+                ['max_length', [':value', 255]],
+                ['regex', [':value', Validator::REGEX_STANDARD_TEXT]], // alpha, number, punctuation, space
+            ],
+            'source' => [
+                ['not_empty'],
+                ['min_length', [':value', 2]],
+                ['max_length', [':value', 255]],
+                ['regex', [':value', Validator::REGEX_STANDARD_TEXT]], // alpha, number, punctuation, space
+            ],
+            'maintainer' => [
+                ['not_empty'],
+                ['min_length', [':value', 2]],
+                ['max_length', [':value', 255]],
+                ['regex', [':value', Validator::REGEX_STANDARD_TEXT]], // alpha, number, punctuation, space
+            ]
+        ], $this->getForeignKeyRules());
```

Comment:
I separated the rules for the keys to help code climate calm down. Then realized I'm still over 25 lines

Reply:
I'm not sure it makes it any more clear either... I don't think you need to take the code climate stuff strictly right now, still just experimenting with it.
Maybe we should just relax that rule a bit?

Reply:
uh. Yea me either. 
I'll change it back today